cmake/Helper.cmake

```diff
@@ -424,94 +424,21 @@ function(wasmedge_setup_boost)
-    FetchContent_Declare(
-      wasmedge_image_libpng
-      URL "https://downloads.sourceforge.net/libpng/libpng-1.6.39.tar.gz"
-      URL_HASH "SHA256=af4fb7f260f839919e5958e5ab01a275d4fe436d45442a36ee62f73e5beb75ba"
-    )
-    FetchContent_MakeAvailable(wasmedge_image_libpng)
-    message(STATUS "Downloading libpng source - done")
-    add_custom_command(
-      OUTPUT ${wasmedge_image_libpng_SOURCE_DIR}/.libs/libpng16.a
-      COMMAND ${CMAKE_COMMAND} -E env CFLAGS=-fPIC ./configure --enable-shared=off
-      COMMAND make
-      WORKING_DIRECTORY ${wasmedge_image_libpng_SOURCE_DIR}
-    )
-    add_custom_target(wasmedgeDepsPNG_target
-      ALL DEPENDS
-      ${wasmedge_image_libpng_SOURCE_DIR}/.libs/libpng16.a
-    )
-    add_dependencies(wasmedgeDepsPNG wasmedgeDepsPNG_target)
-    set_target_properties(wasmedgeDepsPNG
-      PROPERTIES
-      IMPORTED_LOCATION ${wasmedge_image_libpng_SOURCE_DIR}/.libs/libpng16.a
-      INTERFACE_INCLUDE_DIRECTORIES ${wasmedge_image_libpng_SOURCE_DIR}
-    )
-  endif()
-endfunction()
-
-function(wasmedge_setup_jpeg)
-  if(TARGET wasmedgeDepsJPEG)
-    return()
-  endif()
-  # setup libpng
-  add_library(wasmedgeDepsJPEG STATIC IMPORTED GLOBAL)
-  if(APPLE)
-    # For MacOS, use the installed libjpeg library.
-    find_package(JPEG REQUIRED)
-    # The find_package will get the shared library. Therefore find the static one.
-    find_library(JPEG_STATIC NAMES libjpeg.a)
-    set_target_properties(wasmedgeDepsJPEG
-      PROPERTIES
-      IMPORTED_LOCATION ${JPEG_STATIC}
-      INTERFACE_INCLUDE_DIRECTORIES "${JPEG_INCLUDE_DIR}"
-    )
-  elseif(UNIX)
-    # Fetch and build libjpeg.
-    include(FetchContent)
-    message(STATUS "Downloading libjpeg source")
-    FetchContent_Declare(
-      wasmedge_image_libjpeg
-      URL "http://ijg.org/files/jpegsrc.v9e.tar.gz"
-      URL_HASH "SHA256=4077d6a6a75aeb01884f708919d25934c93305e49f7e3f36db9129320e6f4f3d"
-    )
-    FetchContent_MakeAvailable(wasmedge_image_libjpeg)
-    message(STATUS "Downloading libjpeg source - done")
-    add_custom_command(
-      OUTPUT ${wasmedge_image_libjpeg_SOURCE_DIR}/.libs/libjpeg.a
-      COMMAND ${CMAKE_COMMAND} -E env CFLAGS=-fPIC ./configure --enable-shared=off
-      COMMAND make
-      WORKING_DIRECTORY ${wasmedge_image_libjpeg_SOURCE_DIR}
-    )
-    add_custom_target(wasmedgeDepsJPEG_target
-      ALL DEPENDS
-      ${wasmedge_image_libjpeg_SOURCE_DIR}/.libs/libjpeg.a
-    )
-    add_dependencies(wasmedgeDepsJPEG wasmedgeDepsJPEG_target)
-    set_target_properties(wasmedgeDepsJPEG
-      PROPERTIES
-      IMPORTED_LOCATION ${wasmedge_image_libjpeg_SOURCE_DIR}/.libs/libjpeg.a
-      INTERFACE_INCLUDE_DIRECTORIES ${wasmedge_image_libjpeg_SOURCE_DIR}
-    )
-  endif()
+  # setup stb_image
+  include(FetchContent)
+  message(STATUS "Downloading stb_image source")
+  FetchContent_Declare(
+    stb
+    GIT_REPOSITORY https://github.com/nothings/stb.git
+    GIT_TAG        master
```

Comment:
Use `master` is unstable, you cannot know the version unless fetching it. We should use a commit or a tag instead.